### PR TITLE
feat: helm enable external secrets

### PR DESCRIPTION
**Pull request by lufinima**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -137,19 +135,89 @@ helm upgrade relyio-galaxy \
...
+
+The `envs` section is an array of objects that contains the `name` of the env var that we want to set in the helm chart and the `key` that is the key of the value in the secret file.
+
+The creation of the external secrets are outside of the scope of this documentation, but for the previous examples we can assume as example of an `ExternalSecret` resource configuration, that would be created prior to the helm chart install, look like this:
+
+```yaml
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: my-vault-secrets
```
Comment (ricardoapl) [inline README.md:185]:
```suggestion
  name: my-external-secrets
```

File: deploy/helm/galaxy/values.yaml [hunk 2]
```diff
@@ -10,24 +10,46 @@ image:
...
+  # eg.:
+  # envs:
+  # - name: "RELY_API_TOKEN"
+  #   key: "api_token"
+  envs: []
+
+# -- The configuration for the integration
+integration:
+  # -- The type of the integration
+  # can be any of the following: pagerduty, github, gitlab, bitbucket, sonarqube, aws, opsgenie, gcp
```
Comment (ricardoapl) [inline deploy/helm/galaxy/values.yaml:34]: I think it's easy to forget to keep this list up to date

Could we point to a documentation link instead?

File: deploy/helm/galaxy/templates/deployment.yaml [hunk 3]
```diff
@@ -37,9 +37,30 @@ spec:
           {{- end }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- if and .Values.externalSecrets.enabled .Values.externalSecrets.allAsEnv}}
+          env:
+            {{- include "galaxy-helm.base.envs" . | nindent 12 }}
```
Comment (ricardoapl) [inline deploy/helm/galaxy/templates/deployment.yaml:42]: Nitpick: couldn't this be kept outside of the if/else block?

It seems to always be set, regardless of what value that condition evaluates to

Comment (lufinima) [reply deploy/helm/galaxy/templates/deployment.yaml:42]: These should always be set that's why

File: deploy/helm/galaxy/values.yaml (hunk 2 above)
Comment (lufinima) [reply deploy/helm/galaxy/values.yaml:34]: Makes sense, I don't think we have a list in the docs yet, but should  be changed latter I gess

Comment (t1agob) [reply deploy/helm/galaxy/values.yaml:34]: We can put this in the readme file and later add it to docs.rely.io. We still haven't covered it there.

**Review by ricardoapl — APPROVED** [review]
Looks good to me, no blockers from my side

**Pull request merged**